project-koku/koku
# More info on what celery tasks are in queue

## Jira Ticket

[COST-####](https://issues.redhat.com/browse/COST-####)

## Description

This change will create a new masu endpoint to provide information on the queued celery tasks.

## Testing

LOCAL TESTING:
1. Spin up koku on this branch, `make docker-up-min`
2. create a test customer `make create-test-customer`
3. spin down just the koku worker `docker-compose stop koku-worker`
4. add a few tasks to different queues using masu api's, a provider uuid will be required to use a few of the examples below.
```
/api/cost-management/v1/download/

/api/cost-management/v1/report_data/?schema=org1234567&start_date=2023-09-29&end_date=2023-09-30&provider_type=AWS-local&provider_uuid=<your_provider_uuid>&queue=summary
/api/cost-management/v1/report_data/?schema=org1234567&start_date=2023-09-29&end_date=2023-09-30&provider_type=AWS-local&provider_uuid=<your_provider_uuid>&queue=priority
/api/cost-management/v1/report_data/?schema=org1234567&start_date=2023-09-29&end_date=2023-09-30&provider_type=AWS-local&provider_uuid=<your_provider_uuid>&queue=download
```
5. Check the new endpoint and see the information for the celery tasks presented: `/api/cost-management/v1/celery_queue_tasks/`


EPHEMERAL TESTING:
1. Checkout ephemeral namespace
7. Deploy branch to ephemeral
8. Create a source
9. Make the replicas for summary/priority zero
10. Port forward the masu service and trigger some resummary tasks.
11. Hit the new masu endpoint: api/cost-management/v1/celery_queue_tasks/
12. See the task info for the tasks in queue that have not been picked up since the workers are spun down.
13. Try it with some filters, you can filter on queue name and task name: 
14. api/cost-management/v1/celery_queue_tasks/?queue=priority
15. api/cost-management/v1/celery_queue_tasks/?queue=priority&task=masu.processor.tasks.update_summary_tables
16. api/cost-management/v1/celery_queue_tasks/?task=masu.processor.tasks.update_summary_tables

## Notes

...
